src/Core/Auth/Models/Data/SsoConfigurationData.cs

```diff
@@ -22,7 +22,25 @@ public string Serialize()
 
     public SsoType ConfigType { get; set; }
 
-    public bool KeyConnectorEnabled { get; set; }
+    public MemberDecryptionType MemberDecryptionType { get; set; }
+
+    /// <summary>
+    /// Legacy property to determine if KeyConnector was enabled.
+    /// Kept for backwards compatibility with old configs that will not have
+    /// the new MemberDecryptionType when deserialized from the database.
+    /// </summary>
+    [Obsolete("Use MemberDecryptionType instead")]
+    public bool KeyConnectorEnabled
+    {
+        get => MemberDecryptionType == MemberDecryptionType.KeyConnector;
```

Comment:
Should we make this set only as well?

Reply:
We probably could; however, I leaned away from completely removing it to avoid removing the property from the serialized JSON that's stored in the database. That way, should we need to rollback the feature, we never end up with broken configs saved in the database (missing the `keyConnectorEnabled` property).